### PR TITLE
Add a wrapper object to instantiate and configure the OTel SDK

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Wrap the OTel SDK and provide an interface to obtain the relevant APIs. Doing this now to prepare for the ability to provide a wrapped interface for creating Logs as well as configure other Processors and Exporters.

## Testing

Existing tests over this.

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/330?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #312** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #313** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #314** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #315** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #316** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #317** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #318** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/318?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #327** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/327?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #328** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/328?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #330** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/330?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/330?utm_source=stack-comment).

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/OpenTelemetrySdk.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Wrapper that instantiates a copy of the OpenTelemetry SDK configured with the appropriate settings and the given components so
+ * the Embrace SDK can hook into its lifecycle. From this, the Embrace SDK can obtain an implementations of the OpenTelemetry API to
+ * create OpenTelemetry primitives that it can use internally or export to any OpenTelemetry Collectors.
+ */
+internal class OpenTelemetrySdk(
+    openTelemetryClock: Clock,
+    spanProcessor: SpanProcessor
+) {
+    private val sdk = OpenTelemetrySdk
+        .builder()
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/OpenTelemetrySdk.kt:20]: @lucaslabari we can set a `SdkLoggerProvider` here and add a method to return the logger instance

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/InitModule.kt [hunk 2]
```diff
@@ -82,18 +80,15 @@ internal class InitModuleImpl(
         SpansSinkImpl()
     }
 
+    private val openTelemetrySdk: OpenTelemetrySdk by singleton {
+        OpenTelemetrySdk(
+            openTelemetryClock = openTelemetryClock,
+            spanProcessor = EmbraceSpanProcessor(EmbraceSpanExporter(spansSink))
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/InitModule.kt:86]: @nelsitoPuglisi We can configure the SpanProcessor to use a composite `SpanExporter` so we can export to multiple destinations

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/OpenTelemetrySdk.kt (hunk 1 above)
Comment (lucaslabari) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/OpenTelemetrySdk.kt:20]: sound good !!!

**Review by lucaslabari — APPROVED** [review]
LGTM

**Pull request merged**